File: tests/php/testcases/GetTreeTest.php
<?
require_once 'base.php';

class GetTestCase extends TestServiceTestCase {

    public function test_null_ret() {
        $r = $this->service->get_tree(0);
        $this->assertEquals($r, null);
    }

    public function test_notset() {
        $r = $this->service->get_tree(1);
        $this->assertEquals($r->value, 1);
        $this->assertEquals($r->left, null);
        $this->assertEquals($r->right, null);
    }

    public function test_null() {
        $r = $this->service->get_tree(2);
        $this->assertEquals($r->value, 2);
        $this->assertEquals($r->left, null);
        $this->assertEquals($r->right, null);
    }

    public function test() {
        $r = $this->service->get_tree(3);
        $this->assertEquals($r->value, 3);
        $this->assertEquals($r->left->value, 4);
        $this->assertEquals($r->left->left, null);
        $this->assertEquals($r->left->right, null);
        $this->assertEquals($r->right->value, 5);
        $this->assertEquals($r->right->left, null);
        $this->assertEquals($r->right->right, null);
    }
}

File: tests/mono/testcases/GetTreeTest.cs
using System;
using NUnit.Framework;

namespace TestCases {

    [TestFixture()]
    public class GetTreeTest : TestServiceTest {

        [Test()]
        public void test_null_ret() {
            Assert.AreEqual(null, service.get_tree(0));
        }

        [Test()]
        public void test_notset() {
            Tree r = service.get_tree(1);
            Assert.AreEqual(1, r.value);
            Assert.AreEqual(null, r.left);
            Assert.AreEqual(null, r.right);
        }

        [Test()]
        public void test_null() {
            Tree r = service.get_tree(2);
            Assert.AreEqual(2, r.value);
            Assert.AreEqual(null, r.left);
            Assert.AreEqual(null, r.right);
        }

        [Test()]
        public void test() {
            Tree r = service.get_tree(3);
            Assert.AreEqual(3, r.value);
            Assert.AreEqual(4, r.left.value);
            Assert.AreEqual(null, r.left.left);
            Assert.AreEqual(null, r.left.right);
            Assert.AreEqual(5, r.right.value);
            Assert.AreEqual(null, r.right.left);
            Assert.AreEqual(null, r.right.right);
        }
    }
}
